components/script/dom/dedicatedworkerglobalscope.rs

```diff
@@ -618,12 +618,19 @@ impl DedicatedWorkerGlobalScope {
 unsafe extern "C" fn interrupt_callback(cx: *mut JSContext) -> bool {
     let in_realm_proof = AlreadyInRealm::assert_for_cx(SafeJSContext::from_ptr(cx));
     let global = GlobalScope::from_context(cx, InRealm::Already(&in_realm_proof));
-    let worker =
-        DomRoot::downcast::<WorkerGlobalScope>(global).expect("global is not a worker scope");
+    let worker = DomRoot::downcast::<WorkerGlobalScope>(global.clone())
+        .expect("global is not a worker scope");
     assert!(worker.is::<DedicatedWorkerGlobalScope>());
 
+    let is_closing = worker.is_closing();
+    if is_closing {
+        // We are closing, ensure no script can run and potentially hang.
+        // For workers, the closing flag itself cancels all tasks.
+        global.microtask_queue().clear();
+    }
```

Comment:
What if we add a check to GlobalScope::perform_a_microtask_checkpoint and ScriptThread::perform_a_microtask_checkpoint instead of adding this logic to each interrupt callback?

Reply:
Good idea._